### PR TITLE
Fix 'Show only distinct rows' in tm_data_table

### LINKED ISSUE #571: [Question]: Should `Show only distinct rows` in tm_data_table append an extra column with counts?
### What is your question?

When working with shinyapps.io distribution of our show-case teal.gallery apps for `longtitudinal` app at  https://genentech.shinyapps.io/nest_longitudinal_main/ I saw that current behavior appends a column with counts, when an user selects to `Show only distinct rows`
(last state of the app of the day of creation of this issue https://github.com/insightsengineering/teal.gallery/blob/5816df5f41e7d2f1124807b5884d65f2315add31/longitudinal/app.R)

image

This is a bit confusing. Either we should just apply `distinct()` over data, or append counts to show duplication counts but with a better name. `Duplicates/Replicates` is more meaningful than `n`

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by m7pr**
# Pull Request

Fixes #571 and #602

## Changes

This is a team-owned version of #983 (by @nissinbo) with review feedback incorporated.

### Fix
- Replace `dplyr::count()` with `dplyr::distinct()` to avoid adding an extra count column `n` when showing distinct rows in `tm_data_table`.
- Use `dplyr::distinct(dataname[variables])` instead of `dplyr::distinct(dataname, dplyr::across(dplyr::all_of(variables)), .keep_all = TRUE)` followed by `data_distinct[variables]`. This approach:
  - Is simpler and more readable
  - Handles the edge case where `variables` is `NULL` safely (`dataname[NULL]` is valid R, whereas `dplyr::all_of(NULL)` would error)

### Test update
- Updated the e2e test expectation to match the new distinct behavior (no extra `n` column, no re-sorting).

### NEWS
- Used the wording suggested in the review.

Co-authored-by: nissinbo <nissinbo@users.noreply.github.com>

### CODE REVIEW

**Review by llrs-roche — COMMENTED** [review]
Looks good, just one comment about tests

**Review by nissinbo — COMMENTED** [review]
Thanks for the review and suggestions. I agree with your comments and the proposed changes.

File: R/tm_data_table.R [hunk 1]
```diff
@@ -320,7 +320,7 @@ srv_data_table <- function(id,
           expr = {
             variables <- vars
             dataframe_selected <- if (if_distinct) {
-              dplyr::count(dataname, dplyr::across(dplyr::all_of(variables)))
+              dplyr::distinct(dataname[variables])
```
[the remark this answers is not in the document]
Comment (m7pr) [reply R/tm_data_table.R:323]: Hey @llrs-roche I added more tests just in general to test this is distinct option
https://github.com/insightsengineering/teal.modules.general/pull/993/commits/81689a9638b1f7014721d944a6c3121e8d63d4ab

File: tests/testthat/test-tm_data_table.R [hunk 2]
```diff
@@ -137,10 +130,96 @@ testthat::describe("tm_g_response module server behavior", {
...
+        mod$server_args,
+        list(id = "test", data = reactive(data_with_dups))
+      ),
+      expr = {
+        session$setInputs(if_distinct = FALSE)
+        session$flushReact()
+        testthat::expect_no_error(session$returned[["test_df"]]())
+      }
+    )
+  })
```
Comment (llrs-roche) [inline tests/testthat/test-tm_data_table.R]: These are duplicated tests, by default `variables_selected = list()`, did you meant to use `variables_selected = NULL` on  `mod  <-` ?

**Review by llrs-roche — APPROVED** [review]
Just one comment to polish this. Great work to simplify this all

I realized during interactive testing that the app doesn't provide a report:
Image
This is a different issue but I wanted to check with you @m7pr if this was intended or we should open a new one.

**Comment by m7pr** [comment]
Hey @llrs-roche yes, the data table doesnt show the code as it's mainly `DT::datatable(data)` : P

File: tests/testthat/test-tm_data_table.R (hunk 2 above)
Comment (llrs-roche) [reply tests/testthat/test-tm_data_table.R]: @m7pr Whenever you have some time remember to resolve this comment as this is close to the finish line.

Comment (m7pr) [reply tests/testthat/test-tm_data_table.R]: yes @llrs-roche I was about to do that

Comment (m7pr) [reply tests/testthat/test-tm_data_table.R]: https://github.com/insightsengineering/teal.modules.general/pull/993/changes/e01e80bc568d4b9ccb92b64e531687ef6faacc2d
